darwin/dataset/utils.py

```diff
@@ -840,3 +845,51 @@ def sanitize_filename(filename: str) -> str:
         filename = filename.replace(char, "_")
 
     return filename
+
+
+def get_external_file_type(storage_key: str) -> str or None:
+    """
+    Returns the type of file given a storage key.
+
+    Parameters
+    ----------
+    storage_key : str
+        The storage key to get the type of file from.
+
+    Returns
+    -------
+    str
+        The type of file.
+    """
+    for extension in SUPPORTED_IMAGE_EXTENSIONS:
+        if storage_key.endswith(extension):
+            return "image"
+    if storage_key.endswith(".pdf"):
+        return "pdf"
+    if storage_key.endswith(".dcm"):
+        return "dicom"
+    for extension in SUPPORTED_VIDEO_EXTENSIONS:
+        if storage_key.endswith(extension):
+            return "video"
+    return None
+
+
+def parse_external_file_path(storage_key: str, preserve_folders: bool) -> str:
+    """
+    Returns the Darwin dataset path given a storage key.
+
+    Parameters
+    ----------
+    storage_key : str
+        The storage key to parse.
+    preserve_folders : bool
+        Whether to preserve folders or place the file in the Dataset root.
+
+    Returns
+    -------
+    str
+        The parsed external file path.
+    """
+    if not preserve_folders:
+        return "/"
+    return "/" + "/".join(storage_key.split("/")[:-1])
```

Comment:
```suggestion
    return "/".join(storage_key.split("/")[:-1]) + "/"
```

Is the `/` on the end of it? as this could give us double `/` in the start of the string, for example:
```python
$ python
>>> storage_key = "/Users/saurabhchopra/.darwin/image.jpg"
>>> "/" + "/".join(storage_key.split("/")[:-1])
'//Users/saurabhchopra/.darwin'
```

Reply:
II don't believe item paths are allowed to end with a `/`

I put a [/](url) at the front for two reasons:
- 1: Storage keys themselves [cannot start with a](https://docs.v7labs.com/docs/registering-items-from-external-storage#the-basics-1) `/`, therefore we shouldn't ever run into doubles
![image](https://github.com/v7labs/darwin-py/assets/124276291/239f4141-17b3-4df2-a092-e3391c183929)
- 2: If the storage key is in the root folder of the bucket, it'll just be filename.jpg (for example), and to preserve the folder structure, path needs to be `/filename`
- 
However, I can imagine users passing storage keys starting with `/` doing this anyway. What do you think about raising an error if this happens?

Reply:
> This file path is case sensitive, cannot start with a forward slash

My bad, I missed reading this part - That makes sense!! 🙏